algo/src/main/java/org/neo4j/graphalgo/similarity/SparseWeightedInput.java

```diff
@@ -0,0 +1,98 @@
+package org.neo4j.graphalgo.similarity;
+
+import org.neo4j.graphalgo.core.utils.Intersections;
+
+class SparseWeightedInput implements WeightedInput {
+    private final int initialSize;
+    private long id;
+    double[] weights;
+    private int count;
+    private RleDecoder rleDecoder;
+
+    public SparseWeightedInput(long id, double[] weights, int initialSize, int nonSkipSize) {
+        this.initialSize = initialSize;
+        this.id = id;
+        this.weights = weights;
+        this.count = nonSkipSize;
+        rleDecoder = new RleDecoder(initialSize);
```

Comment:
do we really need one decoder per SparseWeightedInput? I thought we could use that from outside if needed?